### ISSUE TITLE
doc: CRD scope requires fixes and improvements

### ISSUE THREAD

**Issue by camilamacedo86** [operator-framework/operator-sdk#3684]
## Bug Report

- we need to improve the doc to make it more clear. See: https://master.sdk.operatorframework.io/docs/building-operators/golang/crds-scope/ 
- we can link k8s reference to describe what CRD cluster and namespaced scope means
- the marker there is wrong. It is: `// +kubebuilder:resource:scope=Cluster`. Example

```go
// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
// +kubebuilder:resource:scope=Cluster

// Memcached2 is the Schema for the memcached2s API
type Memcached2 struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   Memcached2Spec   `json:"spec,omitempty"`
	Status Memcached2Status `json:"status,omitempty"`
}

```

- Also, we ought to tell to the users that to generated Cluster scoped CRD's the can use the flag `--namespaced=false` E.g 

> $ operator-sdk create api --group cache --version v1 --kind Memcached2 --resource=true --controller=true  --namespaced=false 

- And then, that by default they are namespaced scope which means that no requires marker at all.

**Comment by estroz** [comment]
Two things need to be done here:
1. Link the [CRD scoped](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition).
2. Documented `--namespaced`.

**Comment by estroz** [comment]
Closed by #4812
